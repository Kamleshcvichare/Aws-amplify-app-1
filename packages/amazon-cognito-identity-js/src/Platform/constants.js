export const Framework = {
	None: '0',
	ReactNative: '1',
};

export const authCategory = 'auth';

// Actions
export const AuthAction = {
	// Standard flow
	SignUp: '1',
	InitiateAuth: '2',
	ConfirmSignUp: '3',
	ResendConfirmationCode: '4',
	GetUser: '5',
	SetUserMFAPreference: '6',
	SetUserSettings: '7',
	AssociateSoftwareToken: '8',
	VerifySoftwareToken: '9',
	RespondToAuthChallenge: '10',
	DeleteUserAttributes: '11',
	DeleteUser: '12',
	UpdateUserAttributes: '13',
	GetUserAttributeVerificationCode: '14',
	VerifyUserAttribute: '15',
	GlobalSignOut: '16',
	RevokeToken: '17',
	ChangePassword: '18',
	ForgotPassword: '19',
	ConfirmForgotPassword: '20',
	ConfirmDevice: '21',
	UpdateDeviceStatus: '22',
	ForgetDevice: '23',
	ListDevices: '24',
	OAuthToken: '25',
};
