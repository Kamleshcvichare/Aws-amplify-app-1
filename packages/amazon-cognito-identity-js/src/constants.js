export const Category = {
	Auth: '1:1',
	API: '1:2',
	DataStore: '1:3',
	Geo: '1:4',
	Interactions: '1:5',
	Notifications: '1:6',
	Predictions: '1:7',
	PubSub: '1:8',
	PushNotification: '1:9',
	Storage: '1:10',
};

export const CategoryAction = {
	AuthSignUp: '2:1',
	AuthSignIn: '2:2',
	APIQuery: '2:3',
	DataStoreJitteredRetry: '2:4',
};

export const UIComponent = {
	Authenticator: '3:1',
	AccountSettings: '3:2',
	MapView: '3:3',
	FileUploader: '3:4',
};

export const Framework = {
	None: '4:0',
	React: '4:1',
	ReactNative: '4:2',
	Next: '4:3',
	Angular: '4:4',
	Angular2Plus: '4:5',
	Vue: '4:6',
};
