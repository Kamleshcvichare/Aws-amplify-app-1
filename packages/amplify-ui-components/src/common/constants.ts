// Namespace
export const AMPLIFY_UI_PREFIX = 'amplify-ui';

// Classnames
export const AMPLIFY_UI_TEXT_INPUT = `${AMPLIFY_UI_PREFIX}-text-input`;
export const AMPLIFY_UI_SCENE = `${AMPLIFY_UI_PREFIX}-scene`;
export const AMPLIFY_UI_SCENE_LOADING = `${AMPLIFY_UI_PREFIX}-scene-loading`;

// Dictionaries

/* 
  The below constants will be moved to an i18n solution.
  Storing here to have a common place to pull from once we pick up that work.
*/
// Username common constants
export const USERNAME_SUFFIX = 'username';
export const USERNAME_LABEL = 'Username *';
export const USERNAME_PLACEHOLDER = 'Enter your username';

// Email common constants
export const EMAIL_SUFFIX = 'email';
export const EMAIL_LABEL = 'Email Address *';
export const EMAIL_PLACEHOLDER = 'amplify@example.com';

// Code common constants
export const CODE_SUFFIX = 'code';
export const CODE_LABEL = 'Verification code';
export const CODE_PLACEHOLDER = 'Enter code';
export const SEND_CODE = 'Send Code';

// Password common constants
export const FORGOT_PASSWORD_TEXT = 'Forgot your password?';
export const RESET_PASSWORD_TEXT = 'Reset password';
export const RESET_YOUR_PASSWORD = 'Reset your password';
export const PASSWORD_SUFFIX = 'password';
export const PASSWORD_LABEL = 'Password *';
export const PASSWORD_PLACEHOLDER = 'Enter your password';
export const BACK_TO_SIGN_IN = 'Back to Sign In';

// Sign In common constants
// TODO: Change to SIGN_IN_HEADER_TEXT
export const HEADER_TEXT = 'Sign into your account';
// TODO: Change to SIGN_IN_SUBMIT_BUTTON_TEXT
export const SUBMIT_BUTTON_TEXT = 'Sign in';
export const CREATE_ACCOUNT_TEXT = 'Create account';
export const NO_ACCOUNT_TEXT = 'No account?';

// Sign Up common constants
export const SIGN_UP_HEADER_TEXT = 'Create a new account';
export const SIGN_UP_SUBMIT_BUTTON_TEXT = 'Create account';
export const HAVE_ACCOUNT_TEXT = 'Have an account?';
export const SIGN_IN_TEXT = 'Sign in';
export const SIGN_UP_USERNAME_PLACEHOLDER = 'Create a username';
export const SIGN_UP_PASSWORD_PLACEHOLDER = 'Create a password';
export const SIGN_UP_PHONE_NUMBER_LABEL = 'Phone Number *';
export const SIGN_UP_PHONE_NUMBER_PLACEHOLDER = '+15555555555';
