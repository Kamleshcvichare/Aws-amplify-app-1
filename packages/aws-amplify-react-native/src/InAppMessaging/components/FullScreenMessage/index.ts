export { default } from './FullScreenMessage';
export { FullScreenMessageProps } from './types';
