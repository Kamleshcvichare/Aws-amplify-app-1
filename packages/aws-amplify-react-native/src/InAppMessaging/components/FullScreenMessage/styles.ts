/*
 * Copyright 2017-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import {
	BORDER_RADIUS_BASE,
	FONT_SIZE_BASE,
	FONT_SIZE_LARGE,
	FONT_WEIGHT_BASE,
	LIGHT_GREY,
	SPACING_EXTRA_LARGE,
	SPACING_LARGE,
	SPACING_MEDIUM,
	SPACING_SMALL,
} from '../constants';
import { getLineHeight } from '../utils';

import { FullScreenMessageStyle } from './types';

export const styles: FullScreenMessageStyle = {
	buttonContainer: {
		backgroundColor: LIGHT_GREY,
		borderRadius: BORDER_RADIUS_BASE,
		flex: 1,
		margin: SPACING_MEDIUM,
		padding: SPACING_LARGE,
	},
	buttonsContainer: {
		flexDirection: 'row',
		justifyContent: 'center',
	},
	buttonText: {
		fontSize: FONT_SIZE_BASE,
		fontWeight: FONT_WEIGHT_BASE,
		lineHeight: getLineHeight(FONT_SIZE_BASE),
		textAlign: 'center',
	},
	container: {
		padding: SPACING_EXTRA_LARGE,
		flex: 1,
	},
	contentContainer: {
		flex: 1,
	},
	header: {
		fontSize: FONT_SIZE_LARGE,
		fontWeight: FONT_WEIGHT_BASE,
		lineHeight: getLineHeight(FONT_SIZE_LARGE),
	},
	iconButton: {
		alignSelf: 'flex-start',
		marginLeft: 'auto',
		marginRight: SPACING_MEDIUM,
	},
	imageContainer: {
		alignItems: 'center',
		marginTop: SPACING_LARGE,
	},
	message: {
		fontSize: FONT_SIZE_BASE,
		fontWeight: FONT_WEIGHT_BASE,
		lineHeight: getLineHeight(FONT_SIZE_BASE),
	},
	textContainer: {
		flex: 1,
		marginVertical: SPACING_LARGE,
		marginHorizontal: SPACING_SMALL,
		paddingLeft: SPACING_MEDIUM,
	},
};
