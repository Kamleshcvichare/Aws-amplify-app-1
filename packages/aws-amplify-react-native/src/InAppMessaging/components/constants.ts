/*
 * Copyright 2017-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

/**
 * Style constants either match or approximate the values used in the Pinpoint console preview.
 * Some values, such as spacing, are slightly different to allow for a more mobile friendly UX
 */

// color
export const BLACK = '#000';
export const LIGHT_GREY = '#e8e8e8';
export const WHITE = '#fff';

// spacing
export const SMALL_SPACING = 4;
export const BASE_SPACING = 8;
export const LARGE_SPACING = 12;
export const EXTRA_LARGE_SPACING = 16;

// button
export const BUTTON_BORDER_RADIUS = 4;

// font
export const BASE_FONT_SIZE = 16;
export const LARGE_FONT_SIZE = 18;
export const BASE_FONT_WEIGHT = '400';

export const LINE_HEIGHT_MULTIPLIER = 1.5;

// icon
export const ICON_BUTTON_SIZE = 20;
export const ICON_BUTTON_HIT_SLOP = 10;

// component specific constants

// BannerMessage
// iOS shadow values
export const BANNER_SHADOW_HEIGHT = 2;
export const BANNER_SHADOW_WIDTH = 2;
export const BANNER_SHADOW_OPACITY = 0.1;
export const BANNER_SHADOW_RADIUS = 2;

// android shadow values
export const BANNER_ELEVATION = 3;

// TODO: delete these once they are no longer needed
// pinpoint max chars 150
export const MAX_MESSAGE =
	'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar,';

// pinpoint max chars 64
export const MAX_HEADER =
	'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum';

// pinpoint max chars 30
export const MAX_BUTTON =
	'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis,';
