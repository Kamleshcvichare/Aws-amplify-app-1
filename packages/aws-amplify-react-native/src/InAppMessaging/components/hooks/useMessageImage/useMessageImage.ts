/*
 * Copyright 2017-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import { useEffect, useState } from 'react';
import { Image } from 'react-native';
import isNull from 'lodash/isNull';

import { ConsoleLogger as Logger } from '@aws-amplify/core';
import { InAppMessageImage, InAppMessageLayout } from '@aws-amplify/notifications';

import { ImageDimensions, UseMessageImage } from './types';
import { getImageDimensions, prefetchNetworkImage } from './utils';

const logger = new Logger('Notifications.InAppMessaging');

/**
 * Handles prefetching and dimension setting for message images
 *
 * @param {image} object - contains image source
 * @param layout - message layout
 * @returns {UseMessageImage} message image dimensions and rendering related booleans
 */

export default function useMessageImage(image: InAppMessageImage, layout: InAppMessageLayout): UseMessageImage {
	const [imageDimensions, setImageDimensions] = useState<ImageDimensions>(null);
	const { src } = image ?? {};

	const hasSetDimensions = !isNull(imageDimensions);
	const hasImage = !!src;

	const imageFetching = hasImage && !hasSetDimensions;
	const renderImage = hasImage && hasSetDimensions;

	useEffect(() => {
		if (hasImage) {
			prefetchNetworkImage(src).then((loadingState) => {
				// get image size once loaded
				if (loadingState === 'loaded') {
					Image.getSize(
						src,
						(width, height) => {
							setImageDimensions(getImageDimensions(height, width, layout));
						},
						(error) => {
							logger.error(`Unable to retrieve size for image: ${error}`);

							// set dimension values to 0 if size retrieval failure
							setImageDimensions({ height: 0, width: 0 });
						}
					);
				}

				// set dimension values to 0 if prefetch failure
				setImageDimensions({ height: 0, width: 0 });
			});
		}
	}, [hasImage, layout, src]);

	return { imageDimensions, imageFetching, renderImage };
}
