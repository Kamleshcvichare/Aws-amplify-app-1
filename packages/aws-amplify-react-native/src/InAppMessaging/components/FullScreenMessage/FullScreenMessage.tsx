/*
 * Copyright 2017-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React from 'react';
import { useMessageProps } from '../hooks';
import MessageWrapper from '../MessageWrapper';

import FullScreenContent from './FullScreenContent';
import { getStyles } from './styles';
import { FullScreenMessageProps } from './types';

export default function FullScreenMessage(props: FullScreenMessageProps) {
	const { styles } = useMessageProps(props, getStyles);

	return (
		<MessageWrapper style={styles.componentWrapper}>
			<FullScreenContent {...props} />
		</MessageWrapper>
	);
}
