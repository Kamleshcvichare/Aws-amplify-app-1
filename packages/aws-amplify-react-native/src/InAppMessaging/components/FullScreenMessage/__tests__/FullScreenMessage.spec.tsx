/*
 * Copyright 2017-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React from 'react';
import TestRenderer from 'react-test-renderer';

import useMessageImage from '../../hooks/useMessageImage';
import { INITIAL_IMAGE_DIMENSIONS } from '../../hooks/useMessageImage/constants';

import FullScreenMessage from '../FullScreenMessage';

jest.mock('../../hooks/useMessageImage');
jest.mock('../../MessageWrapper', () => 'MessageWrapper');
jest.mock('../FullScreenContent', () => 'FullScreenContent');

const baseProps = { layout: 'FULL_SCREEN' as const };

const mockUseMessageImage = useMessageImage as jest.Mock;

describe('FullScreenMessage', () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	it('renders as expected', () => {
		mockUseMessageImage.mockReturnValueOnce({
			hasRenderableImage: false,
			imageDimensions: INITIAL_IMAGE_DIMENSIONS,
			isImageFetching: false,
		});

		const renderer = TestRenderer.create(<FullScreenMessage {...baseProps} />);

		expect(renderer.toJSON()).toMatchSnapshot();
	});
});
