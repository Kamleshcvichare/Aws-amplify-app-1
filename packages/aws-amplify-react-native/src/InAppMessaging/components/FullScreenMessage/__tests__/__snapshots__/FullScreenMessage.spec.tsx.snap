// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenMessage renders as expected 1`] = `
<MessageWrapper
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {},
      Object {},
      Object {},
    ]
  }
>
  <FullScreenContent
    layout="FULL_SCREEN"
  />
</MessageWrapper>
`;
